src/main/java/tree/TargetTreeNode.kt

```diff
@@ -1,9 +1,17 @@
 package tree
 
-interface TargetTreeNode : Comparable<TargetTreeNode> {
-    fun moveRight(): TargetTreeNode
-    fun moveLeft(): TargetTreeNode
-    fun moveUp(): TargetTreeNode
-    fun moveDown(): TargetTreeNode
-    val name: String
+interface TargetTreeNode {
```

Comment:
Can we make this an abstract class. Instead of having the `get*()` methods we can have read only properties using `val`.
![](https://media.giphy.com/media/l0HlBaC5Kp9cXZDTa/giphy.gif)

Reply:
Yes, some of these changes were made, as addressed in previous requests.